docs/deployment/automated-local.rst

```diff
@@ -0,0 +1,227 @@
+.. include:: ../global.rst.inc
+.. highlight:: none
+
+Automated backups to a local hard drive
+=======================================
+
+This guide shows how to automate backups to a hard drive directly connected
+to your computer. If a backup hard drive is connected, backups are automatically
+started, and the drive shut-down and disconnected when they are done.
+
+This guide is written for a Linux-based operating system and makes use of
+systemd and udev.
+
+Overview
+--------
+
+An udev rule is created to trigger on the addition of block devices. The rule contains a tag
+that triggers systemd to start a oneshot service. The oneshot service executes a script in
+the standard systemd service environment, which automatically captures stdout/stderr and
+logs it to the journal.
+
+The script mounts the added block device, if it is a registered backup drive, and creates
+backups on it. When done, it optionally unmounts the file system and spins the drive down,
+so that it may be physically disconnected.
+
+Configuring the system
+----------------------
+
+First, create the ``/etc/backups`` directory (as root).
+All configuration goes into this directory.
+
+Then, create ``etc/backups/40-backup.rules`` with the following content (all on one line)::
+
+    ACTION=="add", SUBSYSTEM=="bdi", DEVPATH=="/devices/virtual/bdi/*",
+    TAG+="systemd", ENV{SYSTEMD_WANTS}="automatic-backup.service"
+
+.. topic:: Finding a more precise udev rule
+
+	If you always connect the drive(s) to the same physical hardware path, e.g. the same
+	eSATA port, then you can make a more precise udev rule.
+
+	Execute ``udevadm monitor`` and connect a drive to the port you intend to use.
+	You should see a flurry of events, find those regarding the `block` subsystem.
+	Pick the event whose device path ends in something similar to a device file name,
+	typically`sdX/sdXY`. Use the event's device path and replace `sdX/sdXY` after the
+	`/block/` part in the path with a star (\*). For example:
+	`DEVPATH=="/devices/pci0000:00/0000:00:11.0/ata3/host2/target2:0:0/2:0:0:0/block/*"`.
+
+	Reboot a few times to ensure that the hardware path does not change: on some motherboards
+	components of it can be random. In these cases you cannot use a more accurate rule,
+	or need to insert additional stars for matching the path.
+
+The "systemd" tag in conjunction with the SYSTEMD_WANTS environment variable has systemd
+launch the "automatic-backup" service, which we will create next, as the
+``/etc/backups/automatic-backup.service`` file:
+
+.. code-block:: ini
+
+    [Service]
+    Type=oneshot
+    ExecStart=/etc/backups/run.sh
+
+Now, create the main backup script, ``/etc/backups/run.sh``. Below is a template,
+modify it to suit your needs (e.g. more backup sets, dumping databases etc.).
+
+.. code-block:: bash
+
+	#!/bin/bash -ue
+	which python borg > /dev/null
```

Comment:
this seems to be designed as guard when needed software is not installed. But it silently terminates the script which is hard to debug.

Also checking for python is likely a bad idea. Because python3 is what borg needs. (I don't think any distribution is that crazy to ship a python3 implementation that is not callable with python3). Also borg might be the one file binary.